sw/device/lib/base/hardened_status_unittest.cc

```diff
@@ -36,5 +36,32 @@ TEST(HardenedStatus, NormalOkIsNotHardenedOk) {
   EXPECT_EQ(hardened_status_ok(OK_STATUS()), kHardenedBoolFalse);
 }
 
+/**
+ * Run `HARDENED_TRY` and return a non-hardened `OK` if it passes.
+ *
+ * @param status status code to try.
+ */
+__attribute__((noinline)) status_t do_hardened_try(status_t status) {
+  HARDENED_TRY(status);
+  return OK_STATUS();
+}
+
+TEST(HardenedStatus, HardenedTryOfNonHardenedOkIsError) {
+  EXPECT_EQ(status_ok(do_hardened_try(OK_STATUS())), false);
```

Comment:
You might want to change this test condition to something more like:
```
EXPECT_EQ(status_err(do_hardened_try(OK_STATUS())), kInternal);
```

This will check that a non-hardened OK gets converted to the expected error value.